tests_end_to_end/xsoar-saas/test_e2e_demisto_sdk_flow_playbook_xsoar_saas.py

```diff
@@ -0,0 +1,154 @@
+        destination_folder=f"{tmpdir}/git/demisto-sdk",
+        sdk_git_branch=DEMISTO_GIT_PRIMARY_BRANCH,
+    )
+
+    repo = Repo(tmpdir)
+    pack, pack_name, source_pack_path = e2e_tests_utils.create_pack(repo)
+    playbook, playbook_name, source_playbook_path, = e2e_tests_utils.create_playbook(pack, pack_name)
+    assert Path(source_playbook_path).exists()
+
+    logger.info(f"Trying to upload playbook from {source_playbook_path}")
+    Uploader(input=source_playbook_path, insecure=True).upload()
+
+    # Preparing updated pack folder
+    e2e_tests_utils.cli(f"mkdir {tmpdir}/Packs/{pack_name}_testsuite")
+
+    logger.info(
+        f"Trying to download the updated playbook from {playbook_name} to {tmpdir}/Packs/{pack_name}_testsuite/Playbooks"
+    )
+    Downloader(
+        output=f"{tmpdir}/Packs/{pack_name}_testsuite",
+        input=(playbook_name,),
+        insecure=True,
+    ).download()
+    dest_playbook_path = Path(
+        f"{tmpdir}/Packs/{pack_name}_testsuite/Playbooks/{playbook_name}.yml"
+    )
+    assert dest_playbook_path.exists()
+
+    logger.info(
+        f"Generating docs (creating a readme file) for playbook {dest_playbook_path}"
+    )
+    generate_playbook_doc.generate_playbook_doc(input_path=str(dest_playbook_path))
+    assert dest_playbook_path.with_name(f"{playbook_name}_README.md").exists()
+
+    logger.info(f"Formating playbook {dest_playbook_path}")
+    with ChangeCWD(pack.repo_path):
+        format_manager(
+            input=str(dest_playbook_path),
+            assume_answer=True,
+        )
+        logger.info(f"Validating playbook {dest_playbook_path}")
+        OldValidateManager(file_path=str(dest_playbook_path)).run_validation()
+
+        logger.info(f"Uploading updated playbook {dest_playbook_path}")
+        Uploader(
+            input=dest_playbook_path,
+            insecure=True,
+        ).upload()
+
+
+def test_e2e_demisto_sdk_flow_playbook_client(tmpdir, verify_ssl: bool = False):
+    """This flow checks:
+        1. Creates a new playbook and uploading it to the machine using an http request.
+        2. Downloads the playbook using demisto-sdk download command.
+        3. Downloads the script CommonServerUserPowerShell using demisto-sdk upload command.
+        4. Generates docs for the playbook using demisto-sdk generate-docs command.
+        5. Formatting the playbook using the demisto-sdk format command.
+        6. Validates the playbook using the demisto-sdk validate command.
+        7. Uploads the playbook using the demisto-sdk upload command.
+        8. Deletes the playbook using an http request.
+    """
+    demisto_client = get_client_from_server_type(verify_ssl=verify_ssl)
+
+    repo = Repo(tmpdir)
+    pack, pack_name, source_pack_path = e2e_tests_utils.create_pack(repo)
+    playbook, playbook_name, dest_playbook_path, = e2e_tests_utils.create_playbook(pack, pack_name)
+
+    try:
+        demisto_client.client.import_playbook(file=dest_playbook_path)
```

Comment:
we can't use the `Uploader` for uplaod the playbook?

Reply:
We want to emulate a playbook that is added through the UI

Reply:
please document that, although I understand your intention I agree with Guy it's not clear, bc later you do use Downloader()